server/src/test/java/org/elasticsearch/common/geo/SpatialPointTests.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.geo;
+
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.Matchers.not;
+
+public class SpatialPointTests extends ESTestCase {
+
+    public void testEqualsAndHashcode() {
```

Comment:
Do we need this class now that we reverted SpatialPoint as concrete class?

Reply:
forget it, its fine,

Reply:
Initially I had removed it, but then realized it still had value, even with GeoPoint and CartesianPoint not sharing a common parent class, so I brought it back.